packages/frontend/src/components/ui/navbar.tsx

```diff
@@ -0,0 +1,141 @@
+import React, { useEffect, useState } from "react";
+import { NavLink } from "react-router-dom";
+import { ConnectWallet } from "../connect-wallet";
+import { Typography } from "@carrot-kpi/ui";
+import type { NavbarLink } from "../../constants";
+import { cva } from "class-variance-authority";
+import Logo from "../../icons/logo";
+import LogoIcon from "../../icons/logo-icon";
+
+const rootStyles = cva(
+    [
+        "flex",
+        "justify-center",
+        "px-4",
+        "xl:px-32",
+        "bg-grid-light",
+        "bg-left-top",
+    ],
+    {
+        variants: {
+            bgColor: {
+                orange: ["bg-orange"],
+                green: ["bg-green"],
+                transparent: ["bg-transparent"],
+            },
+        },
+    },
+);
+
+const navLinkStyles = cva(
+    [
+        "w-full",
+        "flex",
+        "items-center",
+        "gap-2",
+        "dark:bg-black",
+        "border",
+        "border-black",
+        "dark:border-white",
+        "h-11",
+        "p-[11px]",
+        "rounded-lg",
+        "cursor-pointer",
+    ],
+    {
+        variants: {
+            highlighted: {
+                true: ["text-white", "bg-black"],
+                false: ["bg-white"],
+            },
+        },
+    },
+);
+
+export interface NavbarProps {
+    bgColor?: "transparent" | "orange" | "green";
+    links?: NavbarLink[];
+}
+
+export const Navbar = ({
+    bgColor = "transparent",
+    links = [],
+}: NavbarProps) => {
+    const [, setOpen] = useState(false);
+
+    useEffect(() => {
+        const resizeObserver = new ResizeObserver((entries) => {
+            const { width } = entries[0].contentRect;
+            if (width > 640) setOpen(false);
+        });
+
+        resizeObserver.observe(document.body);
+        return () => {
+            // eslint-disable-next-line react-hooks/exhaustive-deps
+            resizeObserver.unobserve(document.body);
+        };
+    });
```

Comment:
I fear a dependency is missing here. Is this hook fired at all without an empty dependencies array?

Reply:
We can actually remove this altogether, fixed in 34e358a7437afad6342fd9a69fb71f1c56d5c4bb.